packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -1027,12 +1027,16 @@ class Node {
   /// One or more reasons why a task was invalidated.
   ///
   /// May be empty if the task was skipped.
-  final Set<InvalidatedReason> invalidatedReasons = <InvalidatedReason>{};
+  final Map<InvalidatedReasonKind, InvalidatedReason> invalidatedReasons = <InvalidatedReasonKind, InvalidatedReason>{};
```

Comment:
The reason I set things up like this is we sometimes discover each file + kind one at a time, so this lets us build up the invalidation context gradually instead of needing to re-order the build system logic to account for itt